### PR TITLE
assistant: Improve HTTP error messages

### PR DESCRIPTION
**Pull request by dsp-ant**
Error messages from HTTP clients are currently
opaqued by two factors:
* We add context on every call layer, opaquing the error message without ever accessing the underlying message.
* Only displaying the message, which effectively is the most recent context on the highest layer.

This commit passes underlying errors more directly to the assistant panel and displays the source if available.

Closes #ISSUE

Release Notes:

- N/A

### CODE REVIEW

File: crates/anthropic/src/anthropic.rs [hunk 1]
```diff
@@ -296,10 +296,7 @@ pub async fn stream_completion_with_rate_limit_info(
         .body(AsyncBody::from(serialized_request))
         .context("failed to construct request body")?;
 
-    let mut response = client
-        .send(request)
-        .await
-        .context("failed to send request to Anthropic")?;
```
Comment (maxdeviant) [inline crates/anthropic/src/anthropic.rs:302]: The `context` is helpful for backtraces, so I’m not sure we want to remove it entirely.

Comment (dsp-ant) [reply crates/anthropic/src/anthropic.rs:302]: Whats the best way to keep the context but bubble up the error? Basically Ive seen instances where the API times out and all a user gets to see is 'streaming completion failed'

**Comment by dsp-ant** [comment]
Okay new attempt @maxdeviant. This is probably a bit overcomplicated, but i change the way we pass errors around. Again this is pretty hacky and happy to be told what the best approach here is from the Zed perspective:

https://github.com/user-attachments/assets/3fd07191-057d-46c5-abee-31c39d679ada

**Comment by maxdeviant** [comment]
@dsp-ant Thanks for you work on this PR!

I'm going to close this in favor of https://github.com/zed-industries/zed/pull/20740, which achieves roughly the same result with fewer changes.

**Pull request closed without merging**